test/simulator/abilities/unaware.js

```diff
@@ -78,4 +78,18 @@ describe('Unaware', function () {
 		battle.commitDecisions();
 		assert.notStrictEqual(pokemon.maxhp - pokemon.hp, damage);
 	});
+
+	it('should not affect opponent\'s self hit damage from confusion', function () {
```

Comment:
This test doesn't actually test this mechanic at all. It will pass without any changes to Unaware at all.